classes/submission/SubmissionDAO.inc.php

```diff
@@ -106,16 +109,84 @@ function _fromRow($row) {
 		$submission->setSubmissionProgress($row['submission_progress']);
 		$submission->setDateSubmitted($this->datetimeFromDB($row['date_submitted']));
 		$submission->setDateStatusModified($this->datetimeFromDB($row['date_status_modified']));
-		$submission->setDatePublished($this->datetimeFromDB($row['date_published']));
 		$submission->setLastModified($this->datetimeFromDB($row['last_modified']));
 		$submission->setLanguage($row['language']);
 		$submission->setCommentsToEditor($row['comments_to_ed']);
-		$submission->setCitations($row['citations']);
-
-		$this->getDataObjectSettings('submission_settings', 'submission_id', $submission->getId(), $submission);
+		$submission->setHideSubmissionRevisions($row['hide_submission_revisions']);
+		
+		$this->getDataObjectSettings('submission_settings', 'submission_id', $submission->getId(), $submission, $submissionRevision, 'submission_revision');
 
 		return $submission;
 	}
+	
+	/**
+	 * Get all revisions for a submission; default settings: 
+	 * 1) Retrieve only the previous revisions, not the current one
+	 * 2) Get only the id's; optionally the title can be retrieved, too
+	 * 3) Default sorting order is ASC
+	 * @param $submissionId int
+	 * @param $contextId int
+	 * @param $showAll boolean false to exclude current revision; true to fetch all revisions including current
+	 * @param $withTitle boolean false to fetch only the IDs of the revisions; true to fetch additionally the title of the submission
+	 * $param $order string
+	 * @return array
+	 */
+	function getSubmissionRevisions($submissionId, $contextId = null, $showAll = false, $withTitle = false, $order = SORT_DIRECTION_ASC) {
+		$params = array((int) $submissionId);
+		if ($contextId) {
+			$params = array_merge($params, array((int) $contextId));
+		}
+		if ($withTitle) {
+			$params = array_merge($params, array(AppLocale::getLocale()));
```

Comment:
What if there is no title in this locale? I think you should consider other locale (as elsewhere) too. I believe always primary locale is considered, but I am not sure what is the correct way for articles -- primary locale or article locale.